entities/capsule.c

```diff
@@ -0,0 +1,135 @@
+#include "capsule.h"
+
+#include "entities_spawner.h"
+#include "entity.h"
+
+SpecificType active_capsule = CAPSULE_EMPTY;
+uint8_t spawned_mini_vaus = 0;
+float shoot_cooldown = 0.0;
+float catch_cooldown = 0.0;
+float balls_velocity = DEFAULT_BALL_VELOCITY;
+struct CatchedBall catched_ball;
+
+void apply_expand_capsule(VAUS *vaus) {
+    update_VAUS_size(vaus, vaus->expand_size + 1);
+}
+
+void update_balls_velocity(float velocity) {
+    if (velocity > 2.0) {
+        balls_velocity = velocity;
+        Balls *balls = get_balls();
+        for (int i = 0; i < balls->current_balls_count; i++) {
+            Ball *ball = &balls->spawned_balls[i];
+            ball->velocity = balls_velocity;
+        }
+    }
+}
+void apply_slow_capsule() {
+    update_active_capsule(CAPSULE_SLOW);
+    update_balls_velocity(balls_velocity / 2);
```

Comment:
Only update velocity once with slow capsule
```suggestion
    if (velocity < 2) {
        velocity = 2;
    }
    balls_velocity = velocity;
    Balls *balls = get_balls();
    for (int i = 0; i < balls->current_balls_count; i++) {
        Ball *ball = &balls->spawned_balls[i];
        ball->velocity = balls_velocity;
    }
}

void apply_slow_capsule() {
    update_active_capsule(CAPSULE_SLOW);
    update_balls_velocity(DEFAULT_BALL_VELOCITY / 2);
```